src/main/java/seedu/address/model/task/Task.java

```diff
@@ -47,23 +51,64 @@ public Task(String taskName, LocalDateTime deadline, Boolean isDone) {
         name = taskName;
         this.deadline = deadline;
         this.isDone = isDone;
+        this.employee = new ArrayList<>();
     }
 
     /**
-     * Returns true if a given string is a valid project name.
+     * Constructs a Task with an assigned employee.
+     * - for loading of tasks from taskhub.json
+     * @param taskName The name of the task.
+     * @param employee The employee that was assigned to the task.
      */
-    public static boolean isValidTask(String testString) {
-        return testString.matches(VALIDATION_REGEX);
+    public Task(String taskName, LocalDateTime deadline, Employee employee) {
+        requireNonNull(taskName);
+        requireNonNull(deadline);
+        requireNonNull(employee);
+        name = taskName;
+        this.deadline = deadline;
+        this.isDone = false;
+        this.employee = Collections.singletonList(employee);
+    }
+
+    /**
+     * Constructs a Task with a pre-determined Done status and assigned employee
+     * - for loading of tasks from taskhub.json.
+     * @param taskName The name of the task.
+     * @param employee The employee that was assigned to the task.
+     */
+    public Task(String taskName, LocalDateTime deadline, Boolean isDone, Employee employee) {
```

Comment:
Perhaps `boolean` primitive type is better than `Boolean` reference type, since `Boolean` can hold null values and this will cause a `NullPointerException` during the auto-boxing. It is better to prevent the "error" during compile-time than catch it during runtime. There is another constructor in this class with the same potential problem.

Reply:
Actually I was just using the existing implementation of the constructor which should have either been done by @timleow who made the class, or @Chandan8186 who handled the marking of tasks. Perhaps they would be better suited to comment on this instead of me, although I do agree with you.